ovn-org/ovn-kubernetes
# unit tests for the functions in the pkg/util/nicstobridge.go file

**- What this PR does and why is it needed**

This PR is about adding unit tests for the functions in the pkg/util/nicstobridge.go file

**- Special notes for reviewers**

The files that are relevant to the current PR for review are:
- go-controller/pkg/util/nicstobridge.go (this file was modified to use the netLinkOps interface methods to make UT easy )
- go-controller/pkg/util/nicstobridge_test.go
- go-controller/pkg/util/ovs_unit_test.go (added a struct defintion to make easy mocking/adding string parameters)

**- How to verify it**

Pull this PR into your workspace
Navigate to go-controller/pkg/util directory
execute below commands
go test -coverprofile cp.out
go tool cover -html=cp.out
select file nicstobridge.go from drop down to verify coverage

**- Description for the changelog**

unit test coverage for functions in pkg/util/nicstobridge.go
